mainmatter/introduction/short_history/index.tex

```diff
@@ -1,4 +1,117 @@
+
+Glides success was in part made possible by Microsoft's involvement in the Talisman project, which took some of resources from the DirectX team.
+The Talisman project was a new 3D architechture that could reduce the memory bandwidth needed by applying tiled rendering \cite{torborg1996talisman}.
+But as other 3D accelerators became more efficient and cheaper, Microsoft cancelled project Talisman and reverted their focus to Direct3D \cite{wikipedia????talisman}.
+The market share of the Windows operating system gave Microsoft the consumer base that the hardware specific \glspl{API} lacked.
+So soon developers started to focus their attention on Direct3D.
+
+\citet{carmack1996plan} from Id Software critized Direct3D for being too verbose, after trying to port Quake to OpenGL and Direct3D.
+So there were some resistance to Direct3D's market dominance, so some developers insisted on using OpenGL.
+And so the OpenGL and Direct3D became the dominant competitors in the 3D rendering 
+
+\subsection{Modern Graphics \arcshort{API}s}
+
+The early days of 3D programming was still very verbose and inflexible. 
+There were using fixed function pipelines instead of shaders, which ment you could not change how the pipeline worked \cite{davidovic2014fixedfunction}.
+Most of the time 
+But in November 2000, Microsoft released their first version of \gls{HLSL} together with DirectX 8 \cite{wikipedia????directx}.
+\Gls{HLSL} is a high level, C like language that specifies how the \gls{GPU} should render a given model.
+It also served as a way to decide which calculations should be done in which render stages in the pipeline.
+
+OpenGL followed with their own shader language \gls{GLSL} in April 2004.
+Like \gls{HLSL}, \gls{GLSL} is a C like language with some extensions to better support operations common to graphics rendering.
+
+Because OpenGL and Direct3D is in direct competition with eachother, and operate on the same hardware, once a technology, or tool has been developed for one, the other usually follows soon after.
+Therefor we look at the general graphics \gls{API} improvements in the following sections.
+
+The first improvement to the rendering pipeline was geometry shaders.
+This shader was in between the vertex and fragment shader and operated on primitives; points, lines, or triangles, and returned zero or more primitives for rasterizing.
+Because the geometry shader could see multiple verticies at once, developers were exited for the oppotunities this would give.
+A good example of using the geometry shader is particle effects, where the specifics of the four corners of the mesh is less important for the developer than the position of the particle.
+So instead, the developer can send the points of the particles to the \gls{GPU}, and have the geometry shader generate the corners in parallel.
```

Comment:
* sammenskriv med forrige sætning
* fjern "so"
* "send the points" => "send the position" jvf. linjen over.
* "... the corners in parallel" lyder som om hvert hjørne bliver lavet parallelt; det er vel hver **partikkel** der menes?